x-pack/plugins/enterprise_search/public/applications/app_search/components/crawler/crawler_overview_logic.ts

```diff
@@ -55,7 +70,27 @@ export const CrawlerOverviewLogic = kea<
       try {
         const response = await http.get(`/api/app_search/engines/${engineName}/crawler`);
         const crawlerData = crawlerDataServerToClient(response as CrawlerDataFromServer);
-        actions.onFetchCrawlerData(crawlerData);
+        actions.onReceiveCrawlerData(crawlerData);
+      } catch (e) {
+        flashAPIErrors(e);
+      }
+    },
+    deleteDomain: async ({ domain }) => {
+      const { http } = HttpLogic.values;
+      const { engineName } = EngineLogic.values;
+
+      try {
+        const response = await http.delete(
+          `/api/app_search/engines/${engineName}/crawler/domains/${domain.id}`,
+          {
+            query: {
+              respond_with: 'crawler_details',
+            },
+          }
+        );
+        const crawlerData = crawlerDataServerToClient(response as CrawlerDataFromServer);
```

Comment:
[Optional] It doesn't look like you have to cast this response - I'd say skip it if we don't need it, since typing is already declared within the helper
```suggestion
        const crawlerData = crawlerDataServerToClient(response);
```

Reply:
I've removed this in 3a331df1c8f

Reply:
Going to go ahead and commit this above line suggestion along with https://github.com/elastic/kibana/pull/101515#discussion_r651394097 just to get it in